File: test-serenity2.Web/Modules/Games/Platforms/PlatformsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace test_serenity2.Games.Columns;

[ColumnsScript("Games.Platforms")]
[BasedOnRow(typeof(PlatformsRow), CheckNames = true)]
public class PlatformsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public int Year { get; set; }
}

File: test-serenity2.Web/Modules/ServerTypes/Games/PlatformsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PlatformsRow } from "./PlatformsRow";

export interface PlatformsColumns {
    Id: Column<PlatformsRow>;
    Name: Column<PlatformsRow>;
    Year: Column<PlatformsRow>;
}

export class PlatformsColumns extends ColumnsBase<PlatformsRow> {
    static readonly columnsKey = 'Games.Platforms';
    static readonly Fields = fieldsProxy<PlatformsColumns>();
}